CIF-905 - AEM CIF Core Component - XF

dplaton opened the pull request:
## Description

Add support for the experience fragments in the demo store:
- create the Experience Fragment page template
- Create the Experience Fragment proxy component
- Create the Experience Fragment proxy page component
- Update the template policy to include the Experience Fragment component

## Related Issue

CIF-905

## Motivation and Context

Allow support for Experience Fragments in the Demo Store

## How Has This Been Tested?

Functional testing

## Screenshots (if appropriate):

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [X] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [X] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

dplaton committed:
CIF-905 - AEM CIF Core Component - XF

Create an Experience Fragment Web template.
Create the Experience Fragment content page component
Create the proxy component.

dplaton committed:
@trivial Escape special characters in HTL templates

herzog31 commented on src/main/archetype/ui.apps/src/main/content/jcr_root/apps/__appsFolderName__/components/content/experiencefragment/v1/experiencefragment/.content.xml:
```diff
@@ -0,0 +1,8 @@
+    jcr:primaryType="cq:Component"
+    jcr:title="Experience Fragment"
+    sling:resourceSuperType="core/wcm/components/experiencefragment/v1/experiencefragment"
+    componentGroup="venia"/>
```
We have a `componentGroupName` placeholder in the archetype for this.

herzog31 commented on src/main/archetype/ui.apps/src/main/content/jcr_root/apps/__appsFolderName__/components/structure/xfpage/.content.xml:
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<jcr:root xmlns:sling="http://sling.apache.org/jcr/sling/1.0" xmlns:cq="http://www.day.com/jcr/cq/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0"
+    jcr:description="Page with Venia specific components for Experience Fragments web variations"
```
Please use `siteName` instead of venia.

cjelger replied:
You forgot to also replace "Venia" in the description

herzog31 commented on src/main/archetype/ui.content/src/main/content/jcr_root/conf/__confFolderName__/settings/wcm/templates/experience-fragment-page/.content.xml:
```diff
@@ -0,0 +1,11 @@
+    <jcr:content
+        cq:lastModified="{Date}2020-01-10T13:53:38.591+02:00"
+        cq:lastModifiedBy="admin"
+        cq:templateType="/conf/venia/settings/wcm/template-types/empty-page-template"
```
Please use `confFolderName`.

herzog31 commented on src/main/archetype/ui.content/src/main/content/jcr_root/conf/__confFolderName__/settings/wcm/templates/experience-fragment-page/.content.xml:
```diff
@@ -0,0 +1,11 @@
+        cq:lastModifiedBy="admin"
+        cq:templateType="/conf/venia/settings/wcm/template-types/empty-page-template"
+        jcr:primaryType="cq:PageContent"
+        jcr:title="Experience Fragment Venia Demo Store"
```
Please use `siteName`.

herzog31 commented on src/main/archetype/ui.content/src/main/content/jcr_root/conf/__confFolderName__/settings/wcm/templates/experience-fragment-page/initial/.content.xml:
```diff
@@ -0,0 +1,17 @@
+    <jcr:content
+        cq:lastModified="{Date}2020-01-10T13:53:38.591+02:00"
+        cq:lastModifiedBy="admin"
+        cq:template="/conf/venia/settings/wcm/templates/experience-fragment-venia-demo-store"
```
Please use `confFolderName`.

herzog31 commented on src/main/archetype/ui.content/src/main/content/jcr_root/conf/__confFolderName__/settings/wcm/templates/experience-fragment-page/initial/.content.xml:
```diff
@@ -0,0 +1,17 @@
+        cq:lastModifiedBy="admin"
+        cq:template="/conf/venia/settings/wcm/templates/experience-fragment-venia-demo-store"
+        jcr:primaryType="cq:PageContent"
+        sling:resourceType="venia/components/structure/xfpage"
```
Please use `appsFolderName`.

herzog31 commented on src/main/archetype/ui.content/src/main/content/jcr_root/conf/__confFolderName__/settings/wcm/templates/experience-fragment-page/policies/.content.xml:
```diff
@@ -0,0 +1,20 @@
+    <jcr:content
+        cq:lastModified="{Date}2020-01-10T12:53:28.602+02:00"
+        cq:lastModifiedBy="admin"
+        cq:policy="venia/components/structure/page/default"
```
Please use `appsFolderName`.

herzog31 commented on src/main/archetype/ui.content/src/main/content/jcr_root/conf/__confFolderName__/settings/wcm/templates/experience-fragment-page/structure/.content.xml:
```diff
@@ -0,0 +1,28 @@
+        cq:deviceGroups="[mobile/groups/responsive]"
+        cq:lastModified="{Date}2018-10-17T10:18:24.970+02:00"
+        cq:lastModifiedBy="admin"
+        cq:template="/conf/venia/settings/wcm/templates/experience-fragment-venia-demo-store"
```
Please use `confFolderName`.

herzog31 commented on src/main/archetype/ui.content/src/main/content/jcr_root/conf/__confFolderName__/settings/wcm/templates/experience-fragment-page/structure/.content.xml:
```diff
@@ -0,0 +1,28 @@
+        cq:lastModifiedBy="admin"
+        cq:template="/conf/venia/settings/wcm/templates/experience-fragment-venia-demo-store"
+        jcr:primaryType="cq:PageContent"
+        sling:resourceType="venia/components/structure/page">
```
Please use `appsFolderName`.

herzog31 reviewed (changes requested):
You missed the archetype placeholders in the xml files.

dplaton committed:
Follow-up on the review:
- use placeholders instead of harcoded names
- remove the versioning folder of the exeperiencefragment component

cjelger commented on src/main/archetype/ui.content/src/main/content/jcr_root/conf/__confFolderName__/settings/wcm/policies/.content.xml:
```diff
@@ -61,7 +61,7 @@
                                     assetMimetype="text/html"
                                     droptarget="experiencefragment"
-                                    resourceType="cq/experience-fragments/editor/components/experiencefragment"
+                                        resourceType="${appsFolderName}/components/content/experiencefragment"
```
Looks like a tab or 4-space identation issue?

cjelger commented on src/main/archetype/ui.content/src/main/content/jcr_root/conf/__confFolderName__/settings/wcm/templates/experience-fragment-page/initial/.content.xml:
```diff
@@ -0,0 +1,17 @@
+    <jcr:content
+        cq:lastModified="{Date}2020-01-10T13:53:38.591+02:00"
+        cq:lastModifiedBy="admin"
+        cq:template="/conf/${confFolderName}/settings/wcm/templates/experience-fragment-venia-demo-store"
```
there is still "venia" hardcoded in the demo-store name. Isn't the name wrong here?... I cannot find this anywhere in your commit, so where is it coming from?

dplaton replied:
It's the name of the template. I'll update the name.

cjelger commented on src/main/archetype/ui.content/src/main/content/jcr_root/conf/__confFolderName__/settings/wcm/templates/experience-fragment-page/structure/.content.xml:
```diff
@@ -0,0 +1,28 @@
+        cq:deviceGroups="[mobile/groups/responsive]"
+        cq:lastModified="{Date}2018-10-17T10:18:24.970+02:00"
+        cq:lastModifiedBy="admin"
+        cq:template="/conf/${confFolderName}/settings/wcm/templates/experience-fragment-venia-demo-store"
```
there is still "venia" hardcoded in the demo-store name

cjelger replied:
Should be `/conf/${confFolderName}/settings/wcm/templates/experience-fragment-page`?

dplaton replied:
Arrrgh, yes it should, I missed this one...

dplaton committed:
CIF-905 - AEM CIF Core Component - XF

Follow-up on the review - fix formatting and replace the hardcoded site name with placeholders

dplaton committed:
Merge branch 'master' into issue/CIF-905

dplaton committed:
CIF-905 - AEM CIF Core Component - XF

Rename a leftover `cq:template` property
